taichi/transforms/remove_assume_in_range.cpp

```diff
@@ -0,0 +1,41 @@
+#include "taichi/ir/ir.h"
+#include "taichi/ir/transforms.h"
+#include "taichi/ir/visitors.h"
+
+TLANG_NAMESPACE_BEGIN
+
+namespace {
+
+// Remove all the assume in range statements.
+// These statements are useless after make_block_local.
+// Their existence harms IR optimization quality.
```

Comment:
Oh, I thought the statement was supplementary and had no effect itself but only side effects of its operands, so I wrote
https://github.com/taichi-dev/taichi/blob/b0b60a7da36ef2fb3a93924ebe8a44b4d2778622/taichi/ir/ir.h#L1006-L1008

But since I understand that `RangeAssumptionStmt` simply returns the range-assumed input, we can directly remove the above lines...